ilastik/applets/dataSelection/datasetInfoEditorWidget.py

```diff
@@ -251,57 +251,38 @@ def _applyTempOpSettingsRealOp(self):
+                    realSlot.setValue(info)
+            except DatasetConstraintError as ex:
+                if not hasattr(ex, 'fixing_dialogs') or not ex.fixing_dialogs:
+                    raise ex
                 msg = (
                     f"Can't use given properties for current dataset, because it violates a constraint of "
                     f"the {ex.appletName} component.\n\n{ex.message}\n\nIf possible, fix this problem by adjusting "
                     f"the applet settings in the next window(s).")
                 QMessageBox.warning(self, "Applet Settings Need Correction", msg)
                 for dlg in ex.fixing_dialogs:
                     dlg()
-                try:
-                    for laneIndex, op in list(self.tempOps.items()):
-                        info = copy.copy(op.Dataset.value)
-                        realSlot = self._op.DatasetGroup[laneIndex][self._roleIndex]
-                        realSlot.setValue(info)
-
-                    # fixed DatasetConstraintError and there are no other errors
-                    ret = None
-                except Exception as ex:
-                    # Maybe we fixed DatasetConstraintError, but there is still an(other) Exception
-                    ret = ex
-
-            if ret is not None:
-                # Try to revert everything back to the previous state
-                try:
-                    for laneIndex, info in list(originalInfos.items()):
-                        realSlot = self._op.DatasetGroup[laneIndex][self._roleIndex]
-                        if realSlot is not None:
-                            realSlot.setValue(info)
-                except Exception:
-                    pass
-            # msg = "Failed to apply your new settings to the workflow " \
-            #       "because they violate a constraint of the {} applet.\n\n".format( ex.appletName ) + \
-            #       ex.message
-            # log_exception(logger, msg, level=logging.INFO)
-            # QMessageBox.warning(self, "Unacceptable Settings", msg)
+
+                for laneIndex, op in list(self.tempOps.items()):
+                    info = copy.copy(op.Dataset.value)
+                    realSlot = self._op.DatasetGroup[laneIndex][self._roleIndex]
+                    realSlot.setValue(info)
         except Exception as ex:
-            ret = ex
-            msg = "Failed to apply dialog settings due to an exception:\n"
-            msg += "{}".format(ex)
+            # Try to revert everything back to the previous state
+            for laneIndex, info in list(originalInfos.items()):
+                realSlot = self._op.DatasetGroup[laneIndex][self._roleIndex]
+                if realSlot is not None:
+                    realSlot.setValue(info)
+
+            msg = f"Failed to apply dialog settings due to an exception:\n{ex}"
             log_exception(logger, msg)
             QMessageBox.critical(self, "Error", msg)
 
-        return ret
```

Comment:
the exception seems to be still expected by `DatasetInfoEditorWidget.accept`. If now, some "unfitting data" (ex. z-depth too small) is loaded, the features are silently switched to 2D. Before there was a fixing dialog popping up...